### PR TITLE
Feature/getAccessKey

### PR DESCRIPTION
**Pull request by AnnRiera**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/blockcoders/near-rpc-providers/blob/main/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

File: src/near-rpc-provider.ts [hunk 1]
```diff
@@ -291,7 +293,7 @@ export class NearRpcProvider extends JsonRpcProvider {
     return params
   }
 
-  async getCode(addressOrName: string | Promise<string>, blockTag: BlockTag): Promise<string> {
+  async getCode(addressOrName: string, blockTag: BlockTag): Promise<string> {
```
Comment (bzuker) [inline src/near-rpc-provider.ts:296]: let's allow people to still use `Promise` here, same as in the `base-provider`. We can do `addressOrName = await addressOrName` in our code to enable that (and similar for `blockTag`)

File: src/near-rpc-provider.ts [hunk 2]
```diff
@@ -354,7 +356,7 @@ export class NearRpcProvider extends JsonRpcProvider {
     }
   }
 
-  async getContractState(addressOrName: string, blockTag: BlockTag) {
+  async getContractState(addressOrName: string, blockTag: BlockTag): Promise<GetStateResponse> {
```
Comment (bzuker) [inline src/near-rpc-provider.ts:359]: similar to above, let's keep the input types consistent so let's allow Promises here as well and `await` them in our code

File: src/near-rpc-provider.ts [hunk 3]
```diff
@@ -372,4 +374,23 @@ export class NearRpcProvider extends JsonRpcProvider {
       })
     }
   }
+
+  async getAccessKey(addressOrName: string, publicKey: string, blockTag: BlockTag): Promise<NearAccessKeyResponse> {
```
Comment (bzuker) [inline src/near-rpc-provider.ts:378]: same as above

File: src/near-rpc-provider.ts [hunk 4]
```diff
@@ -372,4 +374,23 @@ export class NearRpcProvider extends JsonRpcProvider {
...
+    let getaccessKeyParams: GetAccessKeyParams = {
```
Comment (bzuker) [inline src/near-rpc-provider.ts:379]:
```suggestion
    let getAccessKeyParams: GetAccessKeyParams = {
```

make sure to also change every other occurance

File: src/responses.ts [hunk 5]
```diff
@@ -181,3 +181,19 @@ export interface GetStateResponse {
   block_hash: string
   id: string
 }
+
+export interface NearAccessKeyResponse {
```
Comment (bzuker) [inline src/responses.ts:185]: I'd call this `GetAccessKeyResponse` or `ViewAccessKeyResponse`

**Review by bzuker — COMMENTED** [review]
looking good, just a couple of comments

File: src/near-rpc-provider.ts (hunk 1 above)
Comment (AnnRiera) [reply src/near-rpc-provider.ts:296]: Mmmh, okay!

File: src/near-rpc-provider.ts (hunk 2 above)
Comment (AnnRiera) [reply src/near-rpc-provider.ts:359]: No problem!

File: src/near-rpc-provider.ts (hunk 3 above)
Comment (AnnRiera) [reply src/near-rpc-provider.ts:378]: Okay!

File: src/responses.ts (hunk 5 above)
Comment (AnnRiera) [reply src/responses.ts:185]: Roger that!

**Review by bzuker — APPROVED** [review]
LGTM

**Pull request merged**